pkg/controller/node/nodecontroller.go

```diff
@@ -402,6 +413,28 @@ func forcefullyDeletePod(c clientset.Interface, pod *api.Pod) error {
 	return err
 }
 
+func (nc *NodeController) updateEvictionRateLimiters(numNodes int) {
+	glog.V(4).Infof("Updating Eviction rate because number of nodes changed to: %v", numNodes)
+	nc.evictorLock.Lock()
+	defer nc.evictorLock.Unlock()
+
+	var deletionEvictionLimiter, terminationEvictionLimiter flowcontrol.RateLimiter
+	if nc.drainTime == 0 {
+		deletionEvictionLimiter = flowcontrol.NewFakeAlwaysRateLimiter()
+		terminationEvictionLimiter = flowcontrol.NewFakeAlwaysRateLimiter()
+	} else {
+		rate := float32(numNodes) / float32(nc.drainTime/time.Second)
+		if (1/rate) < float32(nc.maxNodeEvictionPeriod) && nc.maxNodeEvictionPeriod != 0 {
```

Comment:
Shouldn't this be `>` ? IIUC `1/rate` is the period between evictions as computed based on drainTime. If that period is *greater* than `maxNodeEvictionPeriod` then you want to set the rate based on `maxNodeEvictionPeriod`